components/size_aware_image.jsx

```diff
@@ -181,6 +196,13 @@ export default class SizeAwareImage extends React.PureComponent {
             ariaLabelImage += ` ${fileInfo.name}`.toLowerCase();
         }
 
+        let imgClassName = this.props.className;
+        if (this.handleLargeImageContainer && this.state.isLargeImage) {
```

Comment:
I'm not sure why this isn't being caught by the linter, but I think this is why your new logic isn't changing anything. This should be `this.props.handleLargeImageContainer`, and you would need to make it so that something sets that prop to true like where it does for the `handleSmallImageContainer` prop

Reply:
Ah gotcha. Still wrapping my head around props. Thanks!

Reply:
Upon further inspection, I don't understand why the original props have been unpacked at line 177 then deleted using `Reflect.deleteProperty()`? Maybe that's why the original code had `this.handleSmallImage` instead of `this.props.handleSmallImage`?

Reply:
The reason we unpack some of the fields and then delete some others is because the unpacked ones are used in this method while others like `enablePublicLink` aren't used here, but we still don't want to pass them on to the img tag that's a child of this. Basically, we want `props` (not `this.props`) to only contain things that the `img` knows how to use.

I don't think the original code actually had `this.handleSmallImageContainer` anywhere, but I do see it using `this.props.handleSmallImageContainer` elsewhere though

Reply:
Also, I found out why the linter isn't complaining about `this.handleSmallImageContainer`, and that's because I didn't realize that this is still a plain JS file and that it hasn't been moved to TypeScript yet. I guess I forgot how easy it is to make that sort of mistake without TS 😛